src/main/java/org/generationcp/commons/util/FileNameGenerator.java

```diff
@@ -0,0 +1,85 @@
+package org.generationcp.commons.util;
+
+import org.generationcp.commons.security.SecurityUtil;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class FileNameGenerator {
+	private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyyMMdd");
+	private static final SimpleDateFormat TIME_FORMAT = new SimpleDateFormat("hhmmss");
+	private static final int MAX_SIZE = 256;
+	private static final int MAX_SIZE_NAME = 200;
+
+	public static String generateFileName(final String origFinalName, final String fileExt, final boolean... isAppendExtension) {
+		final String extension = fileExt.contains(".") ? fileExt : "." + fileExt;
+		final boolean isAppend = isAppendExtension.length == 1 ? isAppendExtension[0]: true;
+		final StringBuilder fileName = new StringBuilder();
+		fileName.append(origFinalName);
+		fileName.append("_");
+		fileName.append(SecurityUtil.getLoggedInUserName());
+		fileName.append("_");
+		fileName.append(FileNameGenerator.getTimeStamp());
+		if (isAppend) {
+			fileName.append(extension);
+		}
+		return FileNameGenerator.truncateIfNecessary(fileName.toString(), FileNameGenerator.getFileNameMaxSize(isAppend, extension));
+	}
+
+	public static String generateFileName(final String origFinalName) {
+
+		final int maxSize;
+		String fileExt = "";
+		String oFName = origFinalName;
+		if (origFinalName.contains(".")) {
+			final String oFileName[] = origFinalName.split("\\.");
+			oFName = oFileName[0];
+			if (oFileName.length >= 2) {
+				// Get the Last Group
+				fileExt = oFileName[oFileName.length - 1];
+			}
+			maxSize = FileNameGenerator.getFileNameMaxSize(true, fileExt);
+		} else {
+			maxSize = FileNameGenerator.MAX_SIZE_NAME;
+		}
+
+		final StringBuilder fileName = new StringBuilder();
```

Comment:
You can extract the appending of username and timestamp to common function:
`
		final StringBuilder fileName = new StringBuilder();
		fileName.append(oFName);
		fileName.append("_");
		fileName.append(SecurityUtil.getLoggedInUserName());
		fileName.append("_");
		fileName.append(FileNameGenerator.getTimeStamp());`